rfqmsg/accept.go

```diff
@@ -0,0 +1,225 @@
+
+	// Add asking price record.
+	records = append(records, TypeRecordAcceptAskingPrice(&q.AskingPrice))
+
+	// Add expiry record.
+	records = append(
+		records, TypeRecordAcceptExpiry(&q.Expiry),
+	)
+
+	// Add signature record.
+	records = append(
+		records, TypeRecordAcceptSig(&q.sig),
+	)
+
+	return records
+}
+
+// Encode encodes the structure into a TLV stream.
+func (q *acceptMsgData) Encode(writer io.Writer) error {
+	stream, err := tlv.NewStream(q.encodeRecords()...)
+	if err != nil {
+		return err
+	}
+	return stream.Encode(writer)
+}
+
+// DecodeRecords provides all TLV records for decoding.
+func (q *acceptMsgData) decodeRecords() []tlv.Record {
+	return []tlv.Record{
+		TypeRecordAcceptID(&q.ID),
+		TypeRecordAcceptAskingPrice(&q.AskingPrice),
+		TypeRecordAcceptExpiry(&q.Expiry),
+		TypeRecordAcceptSig(&q.sig),
+	}
+}
+
+// Decode decodes the structure from a TLV stream.
+func (q *acceptMsgData) Decode(r io.Reader) error {
+	stream, err := tlv.NewStream(q.decodeRecords()...)
+	if err != nil {
+		return err
+	}
+	return stream.Decode(r)
+}
+
+// Accept is a struct that represents an accepted quote message.
+type Accept struct {
+	// Peer is the peer that sent the quote request.
+	Peer route.Vertex
+
+	// AssetAmount is the amount of the asset that the accept message
+	// is for.
+	AssetAmount uint64
+
+	// acceptMsgData is the message data for the quote accept message.
+	acceptMsgData
+}
+
+// NewAcceptFromRequest creates a new instance of a quote accept message given
+// a quote request message.
+func NewAcceptFromRequest(request Request, askingPrice lnwire.MilliSatoshi,
+	expiry uint64) Accept {
+
+	var sig [64]byte
+
+	return Accept{
+		Peer:        request.Peer,
+		AssetAmount: request.AssetAmount,
+		acceptMsgData: acceptMsgData{
+			ID:          request.ID,
+			AskingPrice: askingPrice,
+			Expiry:      expiry,
+			sig:         sig,
+		},
+	}
+}
+
+// NewAcceptFromWireMsg instantiates a new instance from a wire message.
+func NewAcceptFromWireMsg(wireMsg WireMessage) (*Accept, error) {
+	// Decode message data component from TLV bytes.
+	var msgData acceptMsgData
+	err := msgData.Decode(bytes.NewReader(wireMsg.Data))
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode quote accept "+
+			"message data: %w", err)
+	}
+
+	return &Accept{
+		Peer:          wireMsg.Peer,
+		acceptMsgData: msgData,
+	}, nil
+}
+
+// ShortChannelId returns the short channel ID of the quote accept.
+func (q *Accept) ShortChannelId() SerialisedScid {
+	// Given valid RFQ message ID, we then define a RFQ short chain ID
+	// (SCID) by taking the last 8 bytes of the RFQ message ID and
+	// interpreting them as a 64-bit integer.
+	scidBytes := q.ID[24:]
+
+	// TODO(ffranr): Am I doing this right?
+	scidInteger := binary.BigEndian.Uint64(scidBytes)
+
+	return SerialisedScid(scidInteger)
+}
+
+// ToWire returns a wire message with a serialized data field.
+//
+// TODO(ffranr): This method should accept a signer so that we can generate a
```

Comment:
For inspiration, you could check out the way we did it in lnd for the gossip wire messages: https://github.com/lightningnetwork/lnd/blob/master/lnwire/channel_update.go#L224-L226

Then follow that to the various call sites.